src/Evolution/Systems/GrMhd/GhValenciaDivClean/Characteristics.cpp

```diff
@@ -0,0 +1,141 @@
+  for (size_t idx = 0; idx < 13; ++idx) {
+    if (achar_speeds.at(idx).size() != num_grid_points) {
+      achar_speeds.at(idx) = DataVector(num_grid_points);
+    }
+  }
+
+  const auto char_speeds_valencia =
+      grmhd::ValenciaDivClean::characteristic_speeds(
+          rest_mass_density, specific_internal_energy, specific_enthalpy,
+          spatial_velocity, lorentz_factor, magnetic_field, lapse, shift,
+          spatial_metric, unit_normal, equation_of_state);
+  for (size_t idx = 0; idx < 9; ++idx) {
+    char_speeds->at(idx) = char_speeds_valencia.at(idx);
+  }
+
+  const auto char_speeds_gh =
+      GeneralizedHarmonic::characteristic_speeds<3, Frame::Inertial>(
+          gh_constraint_gamma1, lapse, shift, unit_normal);
+  for (size_t idx = 0; idx < 4; ++idx) {
+    char_speeds->at(9 + idx) = char_speeds_gh.at(idx);
+  }
+}
+
+template <size_t ThermodynamicDim>
+std::array<DataVector, 13> characteristic_speeds(
+    const Scalar<DataVector>& rest_mass_density,
+    const Scalar<DataVector>& specific_internal_energy,
+    const Scalar<DataVector>& specific_enthalpy,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& spatial_velocity,
+    const Scalar<DataVector>& lorentz_factor,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& magnetic_field,
+    const Scalar<DataVector>& lapse, const tnsr::I<DataVector, 3>& shift,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric,
+    const tnsr::i<DataVector, 3>& unit_normal,
+    const EquationsOfState::EquationOfState<true, ThermodynamicDim>&
+        equation_of_state,
+    const Scalar<DataVector>& gh_constraint_gamma1) noexcept {
+  auto char_speeds = make_with_value<typename Tags::CharacteristicSpeeds::type>(
+      get(lapse), 0.);
+  characteristic_speeds(make_not_null(&char_speeds), rest_mass_density,
+                        specific_internal_energy, specific_enthalpy,
+                        spatial_velocity, lorentz_factor, magnetic_field, lapse,
+                        shift, spatial_metric, unit_normal, equation_of_state,
+                        gh_constraint_gamma1);
+  return char_speeds;
+}
+
+void Tags::ComputeLargestCharacteristicSpeed::function(
+    const gsl::not_null<double*> speed,
+    const Scalar<DataVector>& gh_constraint_gamma1,
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& shift,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric) noexcept {
+  double max_gh_speed = std::numeric_limits<double>::signaling_NaN();
+  GeneralizedHarmonic::Tags::ComputeLargestCharacteristicSpeed<
+      3, Frame::Inertial>::function(make_not_null(&max_gh_speed),
+                                    gh_constraint_gamma1, lapse, shift,
+                                    spatial_metric);
+  // note: the 1.0 is an approximation valid only in flat spacetime. For
+  // better CFL estimates, this should be updated to calculate the
+  // characteristic speed based on the lapse and shift.
```

Comment:
Is this the same approximation mentioned in the ValenciaDivClean system?  If so, would it be correct to call the function from that system here and pick up the better version automatically if that system is updated?